clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -75,6 +82,8 @@
 
     /* max time in ms for the producer to wait for acknowledgement from server*/
     private final int requestTimeoutMs;
+    
+    private final Time time;
```

Comment:
Yeah, I was hoping that we would be able remove the various `long now` parameters. @enothereska, is avoiding an increase in the number of `time.milliseconds()` calls the reason why you didn't do that?